submit-api/migrations/versions/3088bdcf44cc_change_internal_staff_document_to_.py

```diff
@@ -0,0 +1,79 @@
+"""Change internal staff document to package relationship
+
+Revision ID: 3088bdcf44cc
+Revises: e635c4f73779
+Create Date: 2025-06-09 12:06:38.372828
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+# revision identifiers, used by Alembic.
+revision = '3088bdcf44cc'
+down_revision = '164704893569'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # First add the package_id column
+    with op.batch_alter_table('internal_staff_documents', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('package_id', sa.Integer(), nullable=True))
+    
+    # Now we can update the data
+    op.execute("""
+        UPDATE internal_staff_documents isd
+        SET package_id = i.package_id
+        FROM items i
+        WHERE isd.item_id = i.id
+    """)
+    
+    # Then add the foreign key and make it non-nullable
+    with op.batch_alter_table('internal_staff_documents', schema=None) as batch_op:
+        batch_op.create_foreign_key(None, 'packages', ['package_id'], ['id'])
+        batch_op.alter_column('package_id', nullable=False)
+        
+        # Finally remove the old item_id column and its constraint
+        batch_op.drop_constraint('internal_staff_documents_item_id_fkey', type_='foreignkey')
+        batch_op.drop_column('item_id')
+
+    with op.batch_alter_table('items', schema=None) as batch_op:
+        batch_op.drop_constraint('items_package_id_fkey', type_='foreignkey')
+        batch_op.create_foreign_key(None, 'packages', ['package_id'], ['id'])
+    # ### end Alembic commands ###
+
+
+def downgrade():
+    # First add the item_id column as nullable
+    with op.batch_alter_table('internal_staff_documents', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('item_id', sa.INTEGER(), autoincrement=False, nullable=True))
+    
+    # Update the data to populate item_id by finding the corresponding item for each internal staff document
+    op.execute("""
+        UPDATE internal_staff_documents isd
+        SET item_id = (
+            SELECT i.id 
+            FROM items i 
+            WHERE i.package_id = isd.package_id 
+            AND i.type_id = (
+                SELECT it.id 
+                FROM item_types it 
+                WHERE it.name = 'EAO Internal Documents'
```

Comment:
there isn't an item type called EAO Internal Documents. I think this code is correct except for the AND part which could just be removed.

Reply:
thinking back on this. I don't think it's actually possible to get the item_id again. Maybe just leave it without repopulating like you initially did it. I'm sorry